test/hotspot/jtreg/compiler/whitebox/RelocateAndDeoptmizeAllNMethods.java

```diff
@@ -0,0 +1,111 @@
+ * questions.
+ */
+
+/*
+ * @test id=Serial
+ * @bug 8316694
+ * @summary Relocates all nmethods and also deoptimizes to confirm no crashes
+ * @library /test/lib /
+ * @modules java.base/jdk.internal.misc java.management
+ *
+ * @requires vm.opt.DeoptimizeALot != true
+ *
+ * @build jdk.test.whitebox.WhiteBox
+ * @run driver jdk.test.lib.helpers.ClassFileInstaller jdk.test.whitebox.WhiteBox
+ * @run main/othervm -Xbootclasspath/a:. -XX:+UnlockDiagnosticVMOptions -XX:+WhiteBoxAPI -XX:+SegmentedCodeCache -XX:+UseSerialGC compiler.whitebox.RelocateAndDeoptmizeAllNMethods
+ */
+
+/*
+ * @test id=Parallel
+ * @bug 8316694
+ * @summary Relocates all nmethods and also deoptimizes to confirm no crashes
+ * @library /test/lib /
+ * @modules java.base/jdk.internal.misc java.management
+ *
+ * @requires vm.opt.DeoptimizeALot != true
+ *
+ * @build jdk.test.whitebox.WhiteBox
+ * @run driver jdk.test.lib.helpers.ClassFileInstaller jdk.test.whitebox.WhiteBox
+ * @run main/othervm -Xbootclasspath/a:. -XX:+UnlockDiagnosticVMOptions -XX:+WhiteBoxAPI -XX:+SegmentedCodeCache -XX:+UseParallelGC compiler.whitebox.RelocateAndDeoptmizeAllNMethods
+ */
+
+/*
+ * @test id=G1
+ * @bug 8316694
+ * @summary Relocates all nmethods and also deoptimizes to confirm no crashes
+ * @library /test/lib /
+ * @modules java.base/jdk.internal.misc java.management
+ *
+ * @requires vm.opt.DeoptimizeALot != true
+ *
+ * @build jdk.test.whitebox.WhiteBox
+ * @run driver jdk.test.lib.helpers.ClassFileInstaller jdk.test.whitebox.WhiteBox
+ * @run main/othervm -Xbootclasspath/a:. -XX:+UnlockDiagnosticVMOptions -XX:+WhiteBoxAPI -XX:+SegmentedCodeCache -XX:+UseG1GC compiler.whitebox.RelocateAndDeoptmizeAllNMethods
+ */
+
+/*
+ * @test id=Shenandoah
+ * @bug 8316694
+ * @summary Relocates all nmethods and also deoptimizes to confirm no crashes
+ * @library /test/lib /
+ * @modules java.base/jdk.internal.misc java.management
+ *
+ * @requires vm.opt.DeoptimizeALot != true
+ *
+ * @build jdk.test.whitebox.WhiteBox
+ * @run driver jdk.test.lib.helpers.ClassFileInstaller jdk.test.whitebox.WhiteBox
+ * @run main/othervm -Xbootclasspath/a:. -XX:+UnlockDiagnosticVMOptions -XX:+WhiteBoxAPI -XX:+SegmentedCodeCache -XX:+UseShenandoahGC compiler.whitebox.RelocateAndDeoptmizeAllNMethods
+ */
+
+/*
+ * @test id=ZGC
+ * @bug 8316694
+ * @summary Relocates all nmethods and also deoptimizes to confirm no crashes
+ * @library /test/lib /
+ * @modules java.base/jdk.internal.misc java.management
+ *
+ * @requires vm.opt.DeoptimizeALot != true
+ *
+ * @build jdk.test.whitebox.WhiteBox
+ * @run driver jdk.test.lib.helpers.ClassFileInstaller jdk.test.whitebox.WhiteBox
+ * @run main/othervm -Xbootclasspath/a:. -XX:+UnlockDiagnosticVMOptions -XX:+WhiteBoxAPI -XX:+SegmentedCodeCache -XX:+UseZGC compiler.whitebox.RelocateAndDeoptmizeAllNMethods
+ */
+
+package compiler.whitebox;
+
+import jdk.test.whitebox.WhiteBox;
+
+public class RelocateAndDeoptmizeAllNMethods {
```

Comment:
Please rewrite the test to the following which more practical:
1. Compile a set of Java methods with C1. There should be a call graph of the methods with one entry point which produces a result.
2. Call the entry point to produce a result.
3. Relocate nmethods.
4. Call the entry point to produce a new result.
5. Compare the results.
6. Deoptimize the nmethods.
7. Check Java methods don't have nmethods.

Do the same for C2.